From 509bbeb479afd94f8336f83dfb277aebc5d70119 Mon Sep 17 00:00:00 2001
From: venkata Jagadeesh <vjagad@codeaurora.org>
Date: Thu, 1 Oct 2015 13:12:03 +0530
Subject: [PATCH 189/296] Sync Preference in UI for new cached device

Use Case: Inquiry results not updated to UI when A2DP
streaming is initiated.

Failure: Inquiry results updated to settings APP but
it's not reflecting in UI.

Steps:
 1. A2DP streaming from DUT.
 2. OPP Rx initiated from remote.
 3. Initiating inquiry from DUT.

Root Cause: No callback to UI to sync the preference
parameter of new cached device in DeviceFoundHandler class.

Fix: Create preference in UI whenever new cached device is added.
---
 .../src/com/android/settingslib/bluetooth/BluetoothEventManager.java    | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/packages/SettingsLib/src/com/android/settingslib/bluetooth/BluetoothEventManager.java b/packages/SettingsLib/src/com/android/settingslib/bluetooth/BluetoothEventManager.java
index f57d02bb92f..478ba41fe19 100755
--- a/packages/SettingsLib/src/com/android/settingslib/bluetooth/BluetoothEventManager.java
+++ b/packages/SettingsLib/src/com/android/settingslib/bluetooth/BluetoothEventManager.java
@@ -218,6 +218,8 @@ public class BluetoothEventManager {
                 cachedDevice = mDeviceManager.addDevice(mLocalAdapter, mProfileManager, device);
                 Log.d(TAG, "DeviceFoundHandler created new CachedBluetoothDevice: "
                         + cachedDevice);
+                // callback to UI to create Preference for new device
+                dispatchDeviceAdded(cachedDevice);
             }
             cachedDevice.setRssi(rssi);
             cachedDevice.setBtClass(btClass);
-- 
2.11.0

